richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -0,0 +1,42 @@
+package org.fxmisc.richtext.model;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+
+/**
+ * Defines the operations which are supported on a specific segment type.
+ *
+ * @param <SEG>
+ * @param <S>
+ */
+public interface SegmentOps<SEG, S> {
+    public int length(SEG seg);
+
+    public char charAt(SEG seg, int index);
+
+    public String getText(SEG seg);
+
+    public SEG subSequence(SEG seg, int start, int end);
+
+    public SEG subSequence(SEG seg, int start);
+
+    public SEG append(SEG seg, String str);
```

Comment:
Getting back to your previous point, what should the method (and other methods) be changed to?